lib/authentication/auth_default.js

```diff
@@ -10,19 +10,53 @@
  * @returns {Object}
  * @constructor
  */
-function AuthDefault(password) {
+function AuthDefault(connectionConfig) {
+  const password = connectionConfig.password;
+  const mfaToken = connectionConfig.mfaToken;
+  const passcode = connectionConfig.getPasscode();
+  const isPasscodeInPassword = connectionConfig.getPasscodeInPassword();
+  
   /**
-   * Update JSON body with password.
-   *
-   * @param {JSON} body
-   *
-   * @returns {null}
-   */
+     * Update JSON body with password or token.
+     *
+     * @param {JSON} body
+     *
+     * @returns {null}
+     */
   this.updateBody = function (body) {
     body['data']['PASSWORD'] = password;
+
+    if (isMFAAuth()) {
+      setMFASessionParams(body);
+    }
   };
 
-  this.authenticate = async function () {};
+  function isMFAAuth() {
+    return ( connectionConfig.getAuthenticator() === 'USERNAME_PASSWORD_MFA' || mfaToken || passcode || isPasscodeInPassword);
+  }
+
+  function setMFASessionParams(body) {
+    body['data']['TOKEN'] = mfaToken;
+    body['data']['AUTHENTICATOR'] = 'USERNAME_PASSWORD_MFA';
```

Comment:
Please use authenticationTypes.MFA_TOKEN_AUTHENTICATOR instead of 'USERNAME_PASSWORD_MFA'

Reply:
Same as above.

Reply:
FYI: I imported the authentication type in the function(class) instead of global because auth_default loads faster than authentication.js. If I import authenticateType from authentication.js top level, the authenticateType is undefined because of the reason above.

Reply:
My advice is to extract authentication types to new file like authentication_types.js or authentication_util.js. WDYT?

Reply:
That is good idea for the future. I changed my code. Please have a look @sfc-gh-pmotacki.